src/Microsoft.AspNetCore.OData/Batch/ODataBatchRequestItem.cs

```diff
@@ -35,19 +36,12 @@ public static async Task SendRequestAsync(RequestDelegate handler, HttpContext c
 
             if (contentIdToLocationMapping != null)
             {
-                string queryString = context.Request.QueryString.HasValue ? context.Request.QueryString.Value : String.Empty;
-                string resolvedRequestUrl = ContentIdHelpers.ResolveContentId(queryString, contentIdToLocationMapping);
+                string displayUrl = context.Request.GetDisplayUrl();
+                string resolvedRequestUrl = ContentIdHelpers.ResolveContentId(displayUrl, contentIdToLocationMapping);
                 if (!string.IsNullOrEmpty(resolvedRequestUrl))
                 {
-                    Uri resolvedUri = new Uri(resolvedRequestUrl, UriKind.RelativeOrAbsolute);
-                    if (resolvedUri.IsAbsoluteUri)
-                    {
-                        context.Request.CopyAbsoluteUrl(resolvedUri);
-                    }
-                    else
-                    {
-                        context.Request.QueryString = new QueryString(resolvedRequestUrl);
-                    }
+                    Uri resolvedUri = new Uri(resolvedRequestUrl, UriKind.Absolute);
```

Comment:
>Absolute [](start = 74, length = 8)

I don't think that we can be sure the Uri returned from ResolveContentId is absolute. In other places that we call ResolveContentId, we create the Uri as UriKind.RelativeOrAbsolute and then, if !resolvedUri.IsAbsoluteUri we create a new uri using the base address.  See, for example, Ln57-61 in ODataEntityReferenceLinkDeserializer.

Reply:
Since this is retrieved from the HttpRequest it should generally be absolute, but added a TryCreate just in case.

---
In reply to: [487336879](https://github.com/OData/WebApi/pull/2232#discussion_r487336879) [](ancestors = 487336879)